std/uni.d

```diff
@@ -8214,6 +8215,162 @@ unittest //12428
     assert(s == "abcdefghij");
 }
 
+
+// generic toUpper/toLower on whole range, returns range
+private auto toCaser(alias indexFn, uint maxIdx, alias tableFn, Range)(Range str)
+    if (isInputRange!Range && isSomeChar!(ElementEncodingType!Range))
+{
+    static if (!is(Unqual!(ElementEncodingType!Range) : const(dchar)))
```

Comment:
Alright, the documentation does mention this behaviour, although in the future it would be better to document the details of the two decoding styles in a centralized location and refer to that instead of attempting to explain all the details in the docs for each specific function (case in point: the docs for this function mentions only two of the four points listed in your comment).